designs/2024-repo-ecosystem-plugin-tests/README.md

```diff
@@ -0,0 +1,183 @@
+Each added plugin adds a risk of breakage, so plugins will only be added after filing a new issue and gaining team consensus.
+
+### Rollout
+
+This RFC expects the added ecosystem CI job to _likely_ consistently pass.
+A CI job will be added to the `eslint/eslint` repository, but will not immediately be a part of `main` branch or PR branch builds.
+To be safe, this RFC proposes rolling out CI job in three steps:
+
+1. On a CI cron job once a day, targeting the `main` branch but not blocking its builds
+2. On the `main` branch only, with failures showing as failures in its builds
+3. On all PRs targeting the `main` branch, alongside existing CI jobs
+
+Each step will replace the previous step.
+Once all three are done, running ecosystem tests will be a standard part of `main` branch and pull request CI along with existing tasks like linting and testing.
+
+Starting with a job separately from `main` ensures that unexpectedly high frequencies of breakages are caught early, without blocking `main` branch builds.
+At least one month should be held between those steps to make sure the job is consistently passing.
+
+## Out of Scope
+
+Automation could be added for at least the filing of issues on plugin failures.
+That does not seem worth the time expenditure given how rarely plugins are expected to fail.
+This RFC's discussion settled on it not being worth it.
+
+Plugins using internal/private ESLint APIs are one of the canonical examples of what this process is meant to flag.
+However, this process intentionally does not include processes for making code changes in those plugins.
+For downstream repositories, this process only proposes how the ESLint team or PR contributors may file issues.
+This RFC's intent is that those repositories will drive changing their uses of ESLint APIs.
+
+## Open Questions
+
+Are there other plugins we should include that satisfy the criteria?
+
+## Help Needed
+
+I expect to implement this change.
+
+## Frequently Asked Questions
+
+### Given ESLint respects semver, why add tests for plugins that are relying on internals?
+
+It's exceedingly difficult to be sure when changes to a large published package break contracts with downstream consumers.
+Even when all packages in an ecosystem are well-tested the way ESLint and its major plugins are, the sheer project size and duration of maintenance make unfortunate edge cases likely to happen.
+
+> [Venerable xkcd "Workflow" comic](https://xkcd.com/1172)
+
+### What if a breakage causes rules to report incorrectly, but doesn't cause `npm test:eslint-compat` to crash?
+
+Checking for incorrect rule reports is not handled by this RFC.
+All recent significant downstream breakages caused rules to fully crash.
+
+Any kind of rule report verification would necessitate ecosystem tests taking a dependency on the specific reports from downstream plugins.
+This RFC does not believe the effort of keeping snapshots of those reports up-to-date as worthwhile.
+
```

Comment:
I'm not sure if this still applies since we'll run a custom `test:eslint-compat` script, so whatever the script is testing the job fails if `test:eslint-compat` returns a non-zero exit code? And I'm guessing a part (if not all) of the tests run by the script will be rule tests which fail if rules report incorrectly, not only on crashes.